src/app/core/import/import-column-mapping/import-column-mapping.service.spec.ts

```diff
@@ -0,0 +1,129 @@
+import { TestBed } from "@angular/core/testing";
+
+import { ImportColumnMappingService } from "./import-column-mapping.service";
+import { ColumnMapping } from "../column-mapping";
+import { Child } from "app/child-dev-project/children/model/child";
+import { School } from "app/child-dev-project/schools/model/school";
+import { EntitySchema } from "app/core/entity/schema/entity-schema";
+
+describe("ImportColumnMappingService", () => {
+  let service: ImportColumnMappingService;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({});
+    service = TestBed.inject(ImportColumnMappingService);
+  });
+
+  it("should be created", () => {
+    expect(service).toBeTruthy();
+  });
+
+  it("should map columns to their respective property names automatically", () => {
+    // setup
+    const testMapping: ColumnMapping[] = [
+      { column: "name" },
+      { column: "phone" },
+      { column: "address" },
+      { column: "some other remarks" },
+    ];
+    const entitySchema: EntitySchema = new Map([
+      ["name", {}],
+      ["phone", {}],
+      ["address", {}],
+      ["some other remarks", {}],
+    ]);
+
+    // run
+    service.automaticallySelectMappings(testMapping, entitySchema);
+
+    // check
+    expect(testMapping).toEqual([
+      { column: "name", propertyName: "name" },
+      { column: "phone", propertyName: "phone" },
+      { column: "address", propertyName: "address" },
+      { column: "some other remarks" },
+    ]);
+  });
+
+  // TODO: test case for different capitalization, etc.
```

Comment:
one further test case to add here (and then implement the logic for it):

```
  it("should map columns also if they match the field label (instead of field id)", () => {
    // setup
    const testMapping: ColumnMapping[] = [
      { column: "phone number" },
      { column: "name" },
      { column: "address" },
      { column: "some other remarks" },
    ];
    // the schema in Aam Digital defines human-readable labels for each field in addition to the keys:
    const entitySchema: EntitySchema = new Map([
      ["name", { label: "Full Name" }],
      ["phone", { label: "Phone Number" }],
      ["address", { label: "Address" }]
    ]);

    // run
    service.automaticallySelectMappings(testMapping, entitySchema);

    // check
    expect(testMapping).toEqual([
      { column: "name", propertyName: "name" },
      { column: "phone", propertyName: "phone" },
      { column: "address", propertyName: "address" },
      { column: "some other remarks" },
    ]);
  });
```